lib/descendDisplay.ml

```diff
@@ -1282,6 +1288,580 @@ let print_aboville conf base max_level p =
 
 let desmenu_print = Perso.interp_templ "desmenu"
 
+(* new descendant tree algorithm *)
+(* based on the work of Jean Vaucher (U of Montreal) *)
+(* - see issue #414 for comments and details *)
+
+(* *****************************************************************
+   L'algo fait un parcours PRE-ORDRE de l'arbre en donnant une colonne
+   initiale "logique" pour chaque personne, par exemple:  X-1 et X+1
+   pour les 2 enfants d'un pere (P) dans la colonne X.
+   Ce "X" initial peut etre augmenté (déplacé à droite) pour 2 raisons:
+
+   a) La case proposée est déjà occupée.  Un tableau global,
+   last_x, garde la position des derniers noeuds dans chaque rangée.
+   On modifie X = max( X, last_x[i]+2)  gardant au moins une colonne
+   vide entre chaque noeud.
+   b) Pour la meme raison, il se peut que les enfants de la personne
+   soient déplacés. ON termine en mettant X a mi-chemin entre les position
+   finale du premier et dernier enfant.
+
+    ***************************************************************** *)
+
+(* one td is [nb_cols * align * content]  see dag2Html.ml *)
+
+(* empty fill over n columns *)
+let td_fill x1 xn = [ (xn - x1, CenterA, TDnothing) ]
+
+(* hbar over xn - x1 columns. First and last are left/right and 50% *)
+let td_hbar x1 xn =
+  match xn - x1 with
+  | 0 -> [ (1, CenterA, TDnothing) ]
+  | 1 -> [ (1, CenterA, TDhr CenterA) ]
+  | _ ->
+      [ (1, LeftA, TDhr LeftA) ]
+      @ [ (xn - x1 - 1, CenterA, TDhr CenterA) ]
+      @ [ (1, RightA, TDhr RightA) ]
+
+(* regular cell, centered, with text as content (may contain |<br>) *)
+let td_cell cols align ip text flags =
+  match align with
+  | "center" -> [ (cols, CenterA, TDitem (ip, Adef.safe text, flags)) ]
+  | "right" -> [ (cols, RightA, TDitem (ip, Adef.safe text, flags)) ]
+  | "left" -> [ (cols, LeftA, TDitem (ip, Adef.safe text, flags)) ]
+  | _ -> [ (0, CenterA, TDitem (ip, Adef.safe text, flags)) ]
+
```

Comment:
you might want to make a type for `align` instead of matching on a string

Reply:
Ok, done (I did not wand to interfere with existing types, but this ok)